subject-utils/scripts/subjectListScraper.ts

```diff
@@ -30,11 +33,18 @@ type BaseURL = string;
  *  Generates a base search url given the year and study period
  * @param year The year to search for
  * @param studyPeriod The study period to search for
+ * @param online Whether the subject is online only
  */
-const getBaseURL = (year: number, studyPeriod: SubjectPeriod): BaseURL =>
+const getBaseURL = (year: number, studyPeriod: SubjectPeriod, online:boolean)  =>{
   // tslint:disable-next-line:max-line-length
-  `https://handbook.unimelb.edu.au/search?types[]=subject&year=${year}&subject_level_type[]=all&study_periods[]=${studyPeriod.toLowerCase()}&area_of_study[]=all&org_unit[]=all&campus_and_attendance_mode[]=all&sort=external_code%7Casc`;
-/**
+  if(online){
+    return `https://handbook.unimelb.edu.au/search?types[]=subject&year=${year}&subject_level_type[]=all&study_periods[]=${studyPeriod.toLowerCase()}&area_of_study[]=all&org_unit[]=all&campus_and_attendance_mode[]=Online&sort=external_code%7Casc`;
+  }
```

Comment:
Code duplication, is there a cleaner way to achieve this?

Maybe pulling out the query parameters out into an array, and constructing the final base URL at once. 
```
const campusAttendanceMode = online ? "Online" ? "all";
const queryParameters = [
'types[]=subject',
`year=${year}`,
 `campus_and_attendance_mode[]=${campusAttendanceMode}` 
 ];
 // Construct final base URL using string utilities (reduce would come in handy here)

```

Reply:
Thanks, I tried to remove the repetition and make it as clean as possible. Let me know if you think the changes might be over-complicating it unnecessarily. 

(check the last commit)

😄